### PR TITLE
Enable changing a field from or to list

### PR DESCRIPTION
**Pull request by do4gr**
The deploy mutation at the moment does not generate the necessary tables / columns when a scalar field switches to/from list.

### CODE REVIEW

File: server/servers/deploy/src/test/scala/com/prisma/deploy/specutils/DeployTestServer.scala [hunk 1]
```diff
@@ -114,10 +116,38 @@ case class DeployTestServer()(implicit dependencies: DeployDependencies) extends
...
       Duration.Inf
     )
     println("Request Result: " + result)
     result
   }
+
+  /**
+    * Deploy a schema and return a project
+    */
+  def deploySchema(project: Project, schema: String): Future[Option[Project]] = {
```
Comment (mavilein) [inline server/servers/deploy/src/test/scala/com/prisma/deploy/specutils/DeployTestServer.scala]: This is some duplication of what we already have in `DeploySpecBase.setupProject`. Can you add an `addProject` here and call those methods from `DeploySpecBase` please?

File: server/servers/deploy/src/test/scala/com/prisma/deploy/specutils/DeployTestServer.scala [hunk 2]
```diff
@@ -114,10 +116,38 @@ case class DeployTestServer()(implicit dependencies: DeployDependencies) extends
         )
         .recover {
           case error: QueryAnalysisError => error.resolveError
+          case e                         => sys.error(e.toString)
```
Comment (mavilein) [inline server/servers/deploy/src/test/scala/com/prisma/deploy/specutils/DeployTestServer.scala]: What is this supposed to do? I think this should just discard the stack trace and we therefore lose information in the exception that is reported in a test.

File: server/servers/deploy/src/test/scala/com/prisma/deploy/database/schema/mutations/DeployMutationSpec.scala [hunk 3]
```diff
@@ -659,6 +656,66 @@ class DeployMutationSpec extends FlatSpec with Matchers with DeploySpecBase {
...
+        |type A {
+        | name: String! @unique
+        | value: Int
+        |}
+      """.stripMargin
+
+    server.query(
+      s"""
+         |mutation {
+         |  deploy(input:{name: "${nameAndStage.name}", stage: "${nameAndStage.stage}", types: ${formatSchema(schema)}}){
```
Comment (mavilein) [inline server/servers/deploy/src/test/scala/com/prisma/deploy/database/schema/mutations/DeployMutationSpec.scala]: We should rely on the new shiny `server.deploySchema` here as well.

File: server/servers/deploy/src/test/scala/com/prisma/deploy/specutils/DeployTestServer.scala (hunk 2 above)
Comment (do4gr) [reply server/servers/deploy/src/test/scala/com/prisma/deploy/specutils/DeployTestServer.scala]: oversight from debugging, I removed it.

File: server/servers/deploy/src/test/scala/com/prisma/deploy/specutils/DeployTestServer.scala (hunk 1 above)
Comment (do4gr) [reply server/servers/deploy/src/test/scala/com/prisma/deploy/specutils/DeployTestServer.scala]: Done

**Pull request merged**